code/modules/vehicles/sealed/mecha/mecha_control_console.dm

```diff
@@ -28,7 +28,7 @@
 	var/list/data = ..()
 
 	data["beacons"] = list()
-	for(var/obj/item/vehicle_tracking_beacon/TR in world)
+	for(var/obj/item/vehicle_tracking_beacon/TR in GLOB.vehicle_tracking_beacons)
```

Comment:
GLOB.vehicle_tracking_beacons is going to only contain /obj/item/vehicle_tracking_beacon anyway, so an `as anything` (or whatever you do for that) is appropriate here